### PR TITLE
I22 serialization

### LINKED ISSUE #22: Serde optimization
## Description
We might want to switch to some kind of binary serialization scheme instead of `json` because serializing the proof might be considerably more performant. Check out `bincode`, `cbor` and `borsh` maybe.

### PR DESCRIPTION
**Pull request by gyozosz**
# Description

Aims to resolve #22.
Changed proof serialization to binary scheme, and the three proposed crates `borsh` was the fastest and most concise. Proof input remained serde serialized so that interactions with typescript remain simple.

### CODE REVIEW

File: tom256/Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,8 @@ crate-type = ["cdylib", "rlib"]
 
 [dependencies]
 bigint = { package = "crypto-bigint", git = "https://github.com/PopcornPaws/crypto-bigint", default-features = false }
+bincode = "1.3"
```
Comment (PopcornPaws) [inline tom256/Cargo.toml:11]: do we still need `bincode` here?

File: tom256/src/lib.rs [hunk 2]
```diff
@@ -28,18 +38,28 @@ pub fn generate_proof(input: JsValue, ring: JsValue) -> Result<JsValue, JsValue>
...
 
-    JsValue::from_serde(&zk_attest_proof).map_err(|e| JsValue::from(e.to_string()))
+    let proof_binary = zk_attest_proof
+        .try_to_vec()
+        .map_err(|e| JsValue::from(e.to_string()))?;
+
+    let proof_output = ProofOutput {
+        guild_id: zk_attest_proof.guild_id,
+        r_point: zk_attest_proof.r_point,
+        proof_binary,
```
Comment (PopcornPaws) [inline tom256/src/lib.rs:48]: do we double serialize `guild_id` and `r_point` here? Is there a better way to do this and not include those two fields in in the binary as well?

Comment (PopcornPaws) [reply tom256/src/lib.rs:48]: But probably it's fine because _hopefully_ those two extra fields do not add too much to the proof size.

**Review by PopcornPaws — CHANGES_REQUESTED** [review]

File: tom256/src/lib.rs (hunk 2 above)
Comment (gyozosz) [reply tom256/src/lib.rs:48]: The size of the binary proof is around 160 KB, so adding another point and a string does not increase it substantially but it is more simple. Also, the interface might still frequently change so I do not think we should overspecialize for the current iteration.

**Review by PopcornPaws — APPROVED** [review]

**Pull request merged**